jupyterhub/handlers/pages.py

```diff
@@ -552,6 +552,62 @@ def sort_key(client):
         self.finish(html)
 
 
+class AcceptShareHandler(BaseHandler):
+    @web.authenticated
+    async def get(self):
+        code = self.get_argument("code")
+        next_url = self.get_argument("next", "")
+        share_code = orm.ShareCode.find(self.db, code=code)
+        if share_code is None:
+            raise web.HTTPError(404, "Share not found or expired")
+        if share_code.owner == self.current_user.orm_user:
+            raise web.HTTPError(403, "You can't share with yourself!")
+
+        scope_descriptions = describe_raw_scopes(
+            share_code.scopes,
+            username=self.current_user.name,
+        )
+
+        html = await self.render_template(
+            'accept-share.html',
+            code=code,
+            owner=share_code.owner,
+            spawner=share_code.spawner,
+            scope_descriptions=scope_descriptions,
+            next_url=next_url,
+        )
```

Comment:
Should this form be informed about if the server is started/ready etc?

Reply:
that's a good idea. I'll see how easy that is to add.

Reply:
Now looks like this if the server isn't running:

<img width="808" alt="Screenshot 2024-02-06 at 12 29 22" src="https://github.com/jupyterhub/jupyterhub/assets/151929/55d9ea10-8fec-4ad6-a138-c86304f2a08c">